### DISCUSSION TITLE
What is the state of SSR in React Native?

### DISCUSSION THREAD

**Discussion by AndreiCalazans** [reactwg/react-native-new-architecture#11] [Q&A] [answered]
This question was motived by [The Fabric post]() including the following statement - "Easier to implement server side rendering for React Native" as a benefit of the new renderer. 

Also, recently [Andrew Clark made a reference to SSR with React Native on Twitter](https://twitter.com/acdlite/status/1489755769876754434). 

What is the state of SSR in React Native within Meta? 

Further, why does Fabric make it easier? From my interpretation nothing changed with regards to the exposed APIs so it is not obvious to me.

**Comment by ShikaSD** [comment, marked as answer]
In general, SSR is something we looked into, but decided to ultimately postpone, focusing on executing the client work ahead of time (e.g. pre-loading/pre-rendering).

One of the main reasons why it wasn't a focus is differences between initial render in web browser and RN:
- With the web browser, server-rendered page will be handled by the browser separately, resulting in more efficient rendering, while many elements (e.g. links) will have a basic level of interactivity. This allows to effectively present semi-interactive pages without hydration, and full interactivity when JS is ready.
- With React Native, the situation is a bit different. We could serialize the `ShadowNode` tree and present it in some way, similar to HTML in web browser, but all elements won't be interactive until you load JS and hydrate the tree. Especially this is important for navigation: from RN perspective it is just a touch event, meaning that it won't be handled before the components are hydrated. You could provide custom native components which hook into navigation to work around this, but at this point it is more effective to just focus on creating JS screens that have faster first render and making sure we can pre-load(or pre-render) "heavier" parts.
